compiler.rb

```diff
@@ -120,8 +120,17 @@
     next
   end
 
-  provider_config = \
+  provider_config_or_array = \
     Provider::Config.parse(provider_yaml_path, product_api, version)
+
+  # TODO(alexstephen): Remove when old overrides are deprecated.
+  if provider_config_or_array.is_a?(Array)
+    product_api, provider_config = provider_config
```

Comment:
Also oops, sorry for the late addition here - can you explain with a comment why we're overriding the `product_api`?

Reply:
Done! I think it'll make it easier to understand that the overrides are changing the entirety of the api object after calling `config.validate` too.